inc/classes/subscriber/third-party/plugins/class-smush-subscriber.php

```diff
@@ -24,47 +41,127 @@ public static function get_subscribed_events() {
+	 *
+	 * @since  3.6
+	 * @author Grégory Viguier
+	 *
+	 * @param Options      $options_api WP Options API instance.
+	 * @param Options_Data $options     WP Rocket Options instance.
+	 */
+	public function __construct( Options $options_api, Options_Data $options ) {
+		$this->options_api = $options_api;
+		$this->options     = $options;
+	}
+
 	/**
 	 * Disable WP Rocket lazyload when activating WP Smush and values are already in the database.
 	 *
 	 * @since  3.4.2
 	 * @author Soponar Cristina
 	 */
 	public function maybe_deactivate_rocket_lazyload() {
-		$lazy_load_option = get_option( rocket_get_constant( 'WP_SMUSH_PREFIX' ) . 'settings' );
-		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+		$enabled = $this->is_smush_lazyload_enabled();
+		$updated = false;
+
+		if ( $enabled['images'] && $this->options->get( 'lazyload' ) ) {
+			$this->options->set( 'lazyload', 0 );
+			$updated = true;
+		}
+
+		if ( $enabled['iframes'] && $this->options->get( 'lazyload_iframes' ) ) {
+			$this->options->set( 'lazyload_iframes', 0 );
+			$updated = true;
+		}
 
-		if ( ! $lazyload ) {
+		if ( ! $updated ) {
 			return;
 		}
 
-		update_rocket_option( 'lazyload', 0 );
+		$this->options_api->set( 'settings', $this->options->get_options() );
 	}
 
 	/**
-	 * Disable WP Rocket lazyload fields if WP Smush lazyload is enabled
+	 * Add "Smush" to the provided array if WP Smush lazyload is enabled for images.
 	 *
 	 * @since  3.4.2
 	 * @author Soponar Cristina
 	 *
-	 * @param array $rocket_maybe_disable_lazyload_plugins Array with plugins which disable lazyload functionality.
-	 * @return string
+	 * @param  array $disable_images_lazyload Array with plugins which disable lazyload functionality.
+	 * @return array                          A list of plugin names.
 	 */
-	public function is_smush_lazyload_active( $rocket_maybe_disable_lazyload_plugins ) {
-		$lazy_load_option = get_option( rocket_get_constant( 'WP_SMUSH_PREFIX' ) . 'settings' );
-		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+	public function is_smush_lazyload_active( array $disable_images_lazyload ) {
+		$enabled = $this->is_smush_lazyload_enabled();
+
+		if ( $enabled['images'] ) {
+			$disable_images_lazyload[] = __( 'Smush', 'rocket' );
+		}
+
+		return $disable_images_lazyload;
+	}
+
+	/**
+	 * Tell if Smush’s lazyload is enabled for each type of content.
+	 *
+	 * @since  3.6
+	 * @author Grégory Viguier
+	 *
+	 * @return array {
+	 *     @var bool $images  True when lazyload is enabled for images. False otherwise.
+	 *     @var bool $iframes True when lazyload is enabled for iframes. False otherwise.
+	 * }
+	 */
+	private function is_smush_lazyload_enabled() {
+		$enabled = [
+			'images'  => false,
+			'iframes' => false,
+		];
+
+		if ( ! class_exists( '\Smush\Core\Settings' ) ) {
+			return $enabled;
+		}
+
+		if ( ! Settings::get_instance()->get( 'lazy_load' ) ) {
```

Comment:
Shouldn't we also add a `method_exist` check, we never know what they could decide to change in the class in the future?